### ISSUE TITLE
Not everything is compiled as position independent code with BUILD_SHARED_LIBS=ON

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1553]
**Describe the bug**
When building with `BUILD_SHARED_LIBS=ON`, not everything is compiled as position independent code.

**To reproduce**
Look at the compiler command line: `-fPIC` is present on most, but not all lines. E.g., it is not present when compiling LAPACK, CXSparse and possibly other sub-parts.

E.g.,

```
/opt/local/bin/ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc  -I../include -I../src -Isrc -I../src/f2c -Isrc/f2c -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.14 -Wno-logical-op-parentheses -std=gnu99 -MD -MT src/lapack/CMakeFiles/blas_vendored.dir/dger.c.o -MF src/lapack/CMakeFiles/blas_vendored.dir/dger.c.o.d -o src/lapack/CMakeFiles/blas_vendored.dir/dger.c.o -c ../src/lapack/dger.c
```

```
/opt/local/bin/ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc  -I../include -I../src/cs -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.14 -Wall -Wextra -pedantic -Werror -Wno-unused-parameter -Wno-sign-compare -Wno-varargs -std=gnu99 -MD -MT src/cs/CMakeFiles/cxsparse_vendored.dir/cs_add.c.o -MF src/cs/CMakeFiles/cxsparse_vendored.dir/cs_add.c.o.d -o src/cs/CMakeFiles/cxsparse_vendored.dir/cs_add.c.o -c ../src/cs/cs_add.c
```

**Version information**
`develop` branch.

**Comment by szhorvat** [comment]
I just added a new Travis build with `BUILD_SHARED_LIBS=1`, and it actually fails:

https://travis-ci.com/github/igraph/igraph/jobs/447670209

```
/usr/bin/ld: src/lapack/CMakeFiles/arpack_vendored.dir/dnaupd.c.o: relocation R_X86_64_32 against `.bss' can not be used when making a shared object; recompile with -fPIC

src/lapack/CMakeFiles/arpack_vendored.dir/dnaupd.c.o: error adding symbols: Bad value

collect2: error: ld returned 1 exit status
```

**Comment by szhorvat** [comment]
Fixed in e37e844956cdcdc98fe6e980898ff1bda1758589
